frontend/src/pages/notebookController/screens/server/StopServerModal.tsx

```diff
@@ -29,7 +34,10 @@ const StopServerModal: React.FC<StopServerModalProps> = ({ notebooksToStop, onNo
       notebooksToStop.map((notebook) => {
         const notebookName = notebook.metadata.name || '';
         if (!notebookName) return Promise.resolve();
-        return stopNotebook();
+        if (!impersonatedUsername) {
+          impersonatedUsername = notebook.metadata.labels['opendatahub.io/user'];
```

Comment:
I am not sure whether it's a good practice to manipulating the prop directly, maybe do something like this?
```
remove -> if (!impersonatedUsername) {...}
return stopNotebook(impersonatedUsername ?? notebook.metadata.labels['opendatahub.io/user'])
```

Reply:
If the same solution still applies then this conversation can be resolved.